snippets/snippets.json

```diff
@@ -1,133 +1,97 @@
+		"body": "entity ${1:ent} is\n\tport (\n\t\t$0\n\t);\nend ${1:ent};",
 		"description": "entity",
 		"scope": "source.vhdl"
 	},
 	"entarch": {
 		"prefix": "entarch",
-		"body": "entity ${1:ent} is\n  port (\n\t${0:clock}\n  ) ;\nend ${1:ent} ;\n\narchitecture ${2:arch} of ${1:ent} is\n\n\n\nbegin\n\n\n\nend architecture ; -- ${2:arch}",
-		"description": "entity architecture",
+		"body": "entity ${1:ent} is\n\tport (\n\t\t$0\n\t);\nend ${1:ent};\narchitecture ${2:arch} of ${1:ent} is\nbegin\nend architecture ${2:arch};",
+		"description": "entity + architecture",
 		"scope": "source.vhdl"
 	},
 	"for": {
 		"prefix": "for",
-		"body": "${1:identifier} : for ${2:i} in ${3:0} to ${4:10} loop\n\t$0\nend loop ; -- ${1:identifier}",
+		"body": "for ${1:i} in ${2:range} loop\n\t$0\nend loop;",
 		"description": "for loop",
 		"scope": "source.vhdl"
 	},
 	"forg": {
 		"prefix": "forg",
-		"body": "${1:identifier} : for ${2:i} in ${3:x} to ${4:y} generate\n\t$0\nend generate ; -- ${1:identifier}",
+		"body": "${1:identifier} : for ${2:i} in ${3:range} generate\n\t$0\nend generate ${1:identifier};",
 		"description": "for generate",
 		"scope": "source.vhdl"
 	},
 	"if": {
 		"prefix": "if",
-		"body": "if ${1:expression} then\n\t$0\nend if ;",
+		"body": "if ${1:expression} then\n\t$0\nend if;",
+		"description": "if",
+		"scope": "source.vhdl"
+	},
+	"ifg": {
+		"prefix": "ifg",
+		"body": "${1:label}: if ${2:expression} generate\n\t$0\nend generate ${1:label};",
 		"description": "if",
 		"scope": "source.vhdl"
 	},
 	"pack": {
 		"prefix": "pack",
-		"body": "package ${1:pkg} is\n\t$0\nend package ;",
+		"body": "package ${1:pkg} is\n\t$0\nend package;",
 		"description": "package",
 		"scope": "source.vhdl"
 	},
-	"pro": {
-		"prefix": "pro",
-		"body": "${1:identifier} : process( ${2:sensitivity_list} )\nbegin\n\t$0\nend process ; -- ${1:identifier}",
-		"description": "process",
-		"scope": "source.vhdl"
-	},
-	"proc": {
-		"prefix": "proc",
-		"body": "procedure $1 (\n\t${2:parameter}\n)\nis\nbegin\n\t${3}\nend procedure;",
-		"description": "procedure",
-		"scope": "source.vhdl"
-	},
-	"s": {
-		"prefix": "s",
-		"body": "signed(${1:x} downto ${2:0}) ;$0",
-		"description": "signed downto",
-		"scope": "source.vhdl"
-	},
-	"sr": {
-		"prefix": "sr",
-		"body": "signed(${1:signal}'range) ;$0",
-		"description": "signed range",
-		"scope": "source.vhdl"
-	},
-	"spro": {
-		"prefix": "spro",
-		"body": "${1:identifier} : process( ${2:clock} )\nbegin\n\tif( rising_edge(${2:clock}) ) then\n\t\t$0\n\tend if ;\nend process ; -- ${1:identifier}",
-		"description": "synchronous process",
-		"scope": "source.vhdl"
-	},
-	"slv": {
-		"prefix": "slv",
-		"body": "std_logic_vector(${1:x} downto ${2:0}) ;$0",
-		"description": "std_logic_vector downto",
+	"packbody": {
+		"prefix": "packbody",
+		"body": "package body ${1:pkg} is\n\t$0\nend package body;",
+		"description": "package",
 		"scope": "source.vhdl"
```

Comment:
regarding the format of the end. Here my prefered format is used. Either way this should follow the same logic.